kong/pdk/vault.lua

```diff
@@ -82,9 +83,8 @@ local function new(self)
   end
 
 
-  local function build_cache_key(name, resource, version)
-    return version and fmt("reference:%s:%s:%s", name, resource, version)
-                    or fmt("reference:%s:%s", name, resource)
+  local function build_cache_key(name, resource, version, hash)
+    return fmt("reference:%s:%s:%s:%s", name, resource, version or "", hash or "")
   end
```

Comment:
I found string.format a LOT less performant than string concatenation - in high load situations at least.